qt/scientific_interfaces/Inelastic/Common/OutputPlotOptionsModel.cpp

```diff
@@ -87,9 +87,14 @@ boost::optional<std::string> checkWorkspaceSpectrumSize(const MatrixWorkspace_co
   return boost::none;
 }
 
-boost::optional<std::string> checkWorkspaceBinSize(const MatrixWorkspace_const_sptr &workspace) {
-  if (workspace->getNumberHistograms() < 2)
-    return "Plot Bins failed: There is only one data point to plot in " + workspace->getName() + ".";
+boost::optional<std::string> checkWorkspaceBinSize(const MatrixWorkspace_const_sptr &workspace,
+                                                   const std::string &caller = "") {
+  if (workspace->getNumberHistograms() < 2) {
+    if (caller != "")
+      return caller + ": Can't plot with less than two histograms in  " + workspace->getName() + ".";
+    else
+      return "Plot Bins failed: There is only one data point to plot in " + workspace->getName() + ".";
```

Comment:
Doing it this way may be problematic. When there's no error the return type is boost::none, but if we return something like "Plot Spectra Failed"+boost::none, there may still be a message, so it will catch as an error. 
I'm thinking we can get rid of the boost::none and maybe pass a bool as a reference, with return type std::string, while the bool it is used as a flag for errors?

Reply:
Ah I see, I forgot it uses `boost::optional`!

`boost::optional` effectively acts like a bool flag when used in an if statement. i.e.

```c++
boost::optional<std::string> message = checkWorkspaceBinSize(...);
if (message) {
  // Enters here if a std::string was returned
} else {
  // Enters here if a boost::none was returned
}
```

There is also an opportunity here to modernise some of the code by replacing `boost::optional` with `std::optional`, and replacing `boost::none` with `std::nullopt` (which will reduce our dependency on the boost package)

Reply:
To then get the `std::string` from inside a `std::optional<std::string>` you can then dereference with:

```c++
boost::optional<std::string> message = checkWorkspaceBinSize(...);
std::string messageInside = *message;
```
but note that dereferencing the message with `*message` when the message is a `std::nullopt` will cause an error